internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -74,7 +80,26 @@ func (r *NumaflowControllerRolloutReconciler) Reconcile(ctx context.Context, req
 		}
 	}
 
-	numaLogger.Info(string(numaflowControllerRollout.Spec.Controller.Version))
+	// TODO: need to generate the correct spec and setup the appropriate meta fields
+	// based on the NumaflowControllerRollout spec.controller.version
+	obj := kubernetes.GenericObject{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "",
+			APIVersion: "",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      numaflowControllerRollout.Name,
+			Namespace: "numaflow-system",
+		},
+		Spec: runtime.RawExtension{},
+	}
+
+	// TODO: what is the plural of this kind?
+	err := kubernetes.UpdateCRSpec(ctx, r.restConfig, &obj, "<plural>")
+	if err != nil {
+		numaLogger.Errorf(err, "failed to apply CR: %v", err)
+		return ctrl.Result{}, err
+	}
```

Comment:
jeez, sorry, my original comment was related to "isbservice" 

yeah, I'd remove all of this stuff. She can put what she needs since she's using gitops-engine